File: cinemaProject/src/main/java/com/example/cinemaProject/service/ReviewService.java
package com.example.cinemaProject.service;

import com.example.cinemaProject.DTO.ClientDTO;
import com.example.cinemaProject.DTO.MovieDTO;
import com.example.cinemaProject.DTO.ReviewDTO;
import com.example.cinemaProject.model.Movie;
import com.example.cinemaProject.model.Review;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public interface ReviewService {
    Review saveReview(Review review);
    List<Review> findByStele(int nr);
    Review updateMovie(Review review,String commentNou);
    void deleteReview(int nr);

    ReviewDTO addReview(String nume, String regizor, int stele, String comentariu,
                        String numeClient, String prenumeClient);

    List<ReviewDTO> getAllReviews();

    MovieDTO getMovieFromReview(Long id);
     ClientDTO getClientFromReview(Long id);
}

File: src/app/service/ReviewService.ts
import {Injectable} from "@angular/core";
import {HttpClient, HttpParams} from "@angular/common/http";
import {Movie} from "../model/Movie";
import {Review} from "../model/Review";

@Injectable({
  providedIn: 'root'
})
export class ReviewService {

  baseURL: string = "http://localhost:8081/review";

  constructor(private httpClient: HttpClient) {
  }

  addReview(nume: any,regizor:any,stele:any,comentariu:any,numeC:any,prenume:any) {
    return this.httpClient.put(this.baseURL+"/addReview?"+"nume="+nume
      +"&regizor="+regizor+"&stele="+stele+"&comentariu="+comentariu+"&numeClient="+numeC+
      "&prenumeClient="+prenume,null);
  }


  getAllReviews() {
    return this.httpClient.get<Review[]>(this.baseURL+"/getAllReviews");
  }

  getMovieFromReview(id: any ) {
    return this.httpClient.get<Movie>(this.baseURL+"/getMovieFromReview?id="+id);
  }
  getClientFromReview(id: any ) {
    return this.httpClient.get<Movie>(this.baseURL+"/getClientFromReview?id="+id);
  }

}
